File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(string s) 
{
    vector<string> strNum = 
    {
        "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"
    };
    
    for(int i=0; i<=9; i++)
    {
        while(s.find(strNum[i]) != string::npos)
        {
            size_t pos = s.find(strNum[i]);
            s.replace(pos, strNum[i].size(), to_string(i));
        }
    }
    cout << s;
    return stol(s);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cs
using System;

public class Solution 
{
    public int solution(string s) 
    {
        string[] numArr = new string[10] 
        {
            "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"
        };
        
        for(int i=0; i<=9; i++)
        {
            while(s.Contains(numArr[i]))
            {
                s = s.Replace(numArr[i], i.ToString());
            }
        }
        return Int32.Parse(s);
    }
}